pkg/server/test/write_assertions.go

```diff
@@ -43,44 +45,70 @@ func TestWriteAssertions(t *testing.T, datastore storage.OpenFGADatastore) {
 	var tests = []writeAssertionsTestSettings{
 		{
 			_name: "writing_assertions_succeeds",
-			request: &openfgav1.WriteAssertionsRequest{
-				StoreId: store,
-				Assertions: []*openfgav1.Assertion{{
+			assertions: []*openfgav1.Assertion{{
+				TupleKey:    tuple.NewTupleKey("repo:test", "reader", "user:elbuo"),
+				Expectation: false,
+			}},
+		},
+		{
+			_name: "writing_assertions_succeeds_when_it_is_not_directly_assignable",
+			assertions: []*openfgav1.Assertion{{
+				TupleKey:    tuple.NewTupleKey("repo:test", "can_read", "user:elbuo"),
+				Expectation: false,
+			}},
+		},
+		{
+			_name: "writing_multiple_assertions_succeeds",
+			assertions: []*openfgav1.Assertion{
+				{
 					TupleKey:    tuple.NewTupleKey("repo:test", "reader", "user:elbuo"),
 					Expectation: false,
-				}},
+				},
+				{
+					TupleKey:    tuple.NewTupleKey("repo:test", "reader", "user:maria"),
+					Expectation: true,
+				},
+				{
+					TupleKey:    tuple.NewTupleKey("repo:test", "reader", "user:jon"),
+					Expectation: false,
+				},
+				{
+					TupleKey:    tuple.NewTupleKey("repo:test", "reader", "user:jose"),
+					Expectation: true,
+				},
 			},
 		},
 		{
-			_name: "writing_assertions_succeeds_when_it_is_not_directly_assignable",
-			request: &openfgav1.WriteAssertionsRequest{
-				StoreId: store,
-				Assertions: []*openfgav1.Assertion{{
+			_name: "writing_multiple_assertions_succeeds_when_it_is_not_directly_assignable",
+			assertions: []*openfgav1.Assertion{
+				{
 					TupleKey:    tuple.NewTupleKey("repo:test", "can_read", "user:elbuo"),
 					Expectation: false,
-				}},
+				},
+				{
+					TupleKey:    tuple.NewTupleKey("repo:test", "can_read", "user:maria"),
+					Expectation: false,
+				},
+				{
+					TupleKey:    tuple.NewTupleKey("repo:test", "can_read", "user:jon"),
+					Expectation: true,
+				},
 			},
 		},
 		{
-			_name: "writing_empty_assertions_succeeds",
-			request: &openfgav1.WriteAssertionsRequest{
-				StoreId:    store,
-				Assertions: []*openfgav1.Assertion{},
-			},
+			_name:      "writing_empty_assertions_succeeds",
+			assertions: []*openfgav1.Assertion{},
 		},
 		{
 			_name: "writing_assertion_with_invalid_relation_fails",
-			request: &openfgav1.WriteAssertionsRequest{
-				StoreId: store,
-				Assertions: []*openfgav1.Assertion{
-					{
-						TupleKey: tuple.NewTupleKey(
-							"repo:test",
-							"invalidrelation",
-							"user:elbuo",
-						),
-						Expectation: false,
-					},
+			assertions: []*openfgav1.Assertion{
+				{
+					TupleKey: tuple.NewTupleKey(
+						"repo:test",
+						"invalidrelation",
+						"user:elbuo",
+					),
+					Expectation: false,
 				},
 			},
 			err: serverErrors.ValidationError(
```

Comment:
Can we move the assertions with `err` to a different test called `TestWriteFailure` so that we don't try to read them?

Reply:
Another ask: can you add more tests covering

<img width="1697" alt="image" src="https://github.com/openfga/openfga/assets/5374887/f26d8407-2044-4203-a74c-4ddfffa2709f">